server/container/guice/jpa-common-guice/src/main/java/org/apache/james/modules/data/JPAConfiguration.java

```diff
@@ -18,67 +18,138 @@
  ****************************************************************/
 package org.apache.james.modules.data;
 
-import org.apache.james.backends.jpa.JPAConstants;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Preconditions;
 
 public class JPAConfiguration {
 
-    public static Builder builder() {
-        return new Builder();
+    public static class Credential {
+        private static final Logger LOGGER = LoggerFactory.getLogger(Credential.class);
+
+        public static Optional<Credential> of(String username, String password) {
+            if (Objects.isNull(username) && Objects.isNull(password)) {
+                LOGGER.debug("username and password are missing, returns no credential by default");
+                return Optional.empty();
+            }
+
+            return Optional.of(new Credential(username, password));
+        }
+
+        private final String username;
+        private final String password;
+
+        private Credential(String username, String password) {
+            Preconditions.checkArgument(Objects.nonNull(username) && Objects.nonNull(password),
```

Comment:
Can we have empty string for username or password?

If no, you can use `Strings.isNullOrEmpty`.

Reply:
I use `StringUtils.isBlank()` because it also ships `StringUtils.isNotBlank()`, very convenience :)